### DISCUSSION TITLE
Read-only Public role for anonymous users

### DISCUSSION THREAD

**Discussion by bluepython508** [apache/superset#18553] [Ideas]
**Is your feature request related to a problem? Please describe.**
The documentation recommends setting `PUBLIC_ROLE_LIKE` to `Gamma` to allow anonymous users to use Superset.
However, the `Gamma` role has permissions that are necessary for named users but cause errors for anonymous users, such as `can write on Chart`. This leaves the Save button for charts enabled, but when clicked gives errors about the user being `undefined`, and when a name is entered, the save button is a no-op.

**Describe the solution you'd like**
A role, managed by `superset init`, that is like `Gamma` but designed for anonymous users and use with `PUBLIC_ROLE_LIKE`.
Ideally, this role would contain all of the `can read` and similar permissions that `Gamma` does, but none of the `can write` and similar permissions.

**Describe alternatives you've considered**
- Creating such a role manually
- Leaving the error messages present

**Additional context**
![2022-01-31-152442_681x437_scrot](https://user-images.githubusercontent.com/16466646/151821458-5afc252f-5db6-44be-80a0-6fe7d124a13f.png)

**Comment by geido** [comment]
Hello @bluepython508 I am not sure if I should mark this issue as a bug or enhancement request at the moment. Would you please provide more information concerning the Superset version that you are running and your configuration so that I can try to reproduce the problem? Any additional information, such as videos showing the issue would be immensely helpful!

**Comment by bluepython508** [comment]
It's probably an enhancement request. 
I'm running Superset 1.4 with some patches applied, though that screenshot was taken in a straight install of superset master. The issue arises when setting `PUBLIC_ROLE_LIKE` to `Gamma`, but that is the only config change necessary to reproduce the screenshot. I set `PUBLIC_ROLE_LIKE` to `Gamma`, then added some datasource access to `Gamma`, and tried to make and save a chart.
I'm not really sure what you're asking for video of.

**Comment by geido** [comment]
OK that makes it clearer. From reading your issue the first time I had the impression there was an actual bug. I'll mark it as an enhancement request, even though, as you correctly wrote, creating the role manually is the best option right now. Thank you!

**Comment by geido** [comment]
Related #18210
